### PR TITLE
[Concept Entry] MySQL user-defined variables

### LINKED ISSUE #4680: [Concept Entry] MySQL user defined variables
### Reason/inspiration (optional)

We would like a new entry on the `User Defined Variables` concept under MySQL. The entry should go in a new file under `docs/content/mysql/concepts/user-defined-variables/user-defined-variables.md`.

The entry should include:

- An introductory paragraph that describes the concept
- A ‘Syntax’ section that includes the syntax for the concept (if applicable)
- An ‘Example’ section that shows the concept in use

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by andersooi**
### Description

- Added new concept entry for MySQL User-Defined Variables

### Issue Solved

Closes #4680 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/mysql/concepts/user-defined-variables/user-defined-variables.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+('Tesla', 1089.25);
+```
+
+```sql
+-- Declare a variable to store the total number of stocks
+SET @total_products = (SELECT COUNT(*) FROM stocks_table);
+
+-- Retrieve the value of the variable
+SELECT @total_products;
+```
```
Comment (Sriparno08) [inline content/mysql/concepts/user-defined-variables/user-defined-variables.md:58]: Modifications:

- Let's change the first code block a bit for uniqueness
- Let's convert the comments of the second and third code block into sentences and insert them before the respective code blocks for a better structure

````suggestion
The following example demonstrates the use of a user-defined variable in MySQL:

```sql
-- Create a table named 'stocks_table'
CREATE TABLE stocks_table (
        name VARCHAR(30),
        price DECIMAL(8, 2)
);
```

Then, some stocks are inserted into the table:

```sql
INSERT INTO stocks_table (name, price)
VALUES
('Alphabet', 589.45),
('Apple', 398.23),
('Nvidia', 1237.78),
('Tesla', 1089.25);
```

Next, a user-defined variable is declared to store the total number of stocks and then used to retrieve the stored result:

```sql
SET @total_products = (SELECT COUNT(*) FROM stocks_table);
SELECT @total_products;
```
````

File: content/mysql/concepts/user-defined-variables/user-defined-variables.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+The output of the SELECT statement will be:
```
Comment (Sriparno08) [inline content/mysql/concepts/user-defined-variables/user-defined-variables.md:60]: Let's use single backticks (``` `...` ```) for 'SELECT':

```suggestion
The output of the `SELECT` statement will be:
```

**Review by Sriparno08 — CHANGES_REQUESTED** [review]
Hey @andersooi, thanks for contributing to Codecademy!

I've listed all the modifications that you need to make in your entry below:

**Review by Sriparno08 — APPROVED** [review]
Looks good for a second review, @andersooi! 👍

File: content/mysql/concepts/user-defined-variables/user-defined-variables.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  - 'Database'
+  - 'MySQL'
+  - 'Values'
+  - 'Variables'
+CatalogContent:
+  - 'learn-sql'
+  - 'paths/analyze-data-with-sql'
+---
+
+In MySQL, **user-defined variables** can be used to store a value in a specific statement and later retrieve it in another statement. These variables can be useful for storing intermediate results, passing data between statements, and enhancing the functionality of MySQL queries. They are also _session-specific_, meaning that their values can only be used within the current session and will not be available in future sessions or stored in the database.
```
Comment (ishg-153) [inline content/mysql/concepts/user-defined-variables/user-defined-variables.md:17]:
```suggestion
In MySQL, **user-defined variables** can be used to store a value in a specific statement and later retrieve it in another statement. These variables can be useful for storing intermediate results, passing data between statements, and enhancing the functionality of MySQL queries. They are also session-specific, meaning their values can only be used within the current session and will not be available in future sessions or stored in the database.
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@andersooi Great entry! Please make this one small change. And then we can move forward to merging this PR.

**Review by ishg-153 — APPROVED** [review]
@andersooi Ready to merge!

**Pull request merged**